CI does not catch some vet errors

See #803

We're hitting this issue: https://github.com/golang/go/issues/9439

    ubuntu@box27:~/.gvm/pkgsets/stable/global/src/github.com/docker/distribution$ go tool vet -v registry/storage/blob_test.go 
    vet: registry/storage/blob_test.go:12:2: could not import github.com/docker/distribution (can't find import: github.com/docker/distribution)
    Checking file registry/storage/blob_test.go
    ubuntu@box27:~/.gvm/pkgsets/stable/global/src/github.com/docker/distribution$ godep go install ./...
    ubuntu@box27:~/.gvm/pkgsets/stable/global/src/github.com/docker/distribution$ go tool vet -v registry/storage/blob_test.go 
    vet: registry/storage/blob_test.go:36:14: undeclared name: NewRegistryWithDriver
    Checking file registry/storage/blob_test.go

Let's add `godep go install ./...` before `go vet` to work around it.

--- a/Makefile
+++ b/Makefile
@@ -39,7 +39,9 @@ ${PREFIX}/bin/dist: version/version.go $(shell find . -type f -name '*.go')
 docs/spec/api.md: docs/spec/api.md.tmpl ${PREFIX}/bin/registry-api-descriptor-template
 	./bin/registry-api-descriptor-template $< > $@
 
-vet:
+# Depends on binaries because vet will silently fail if it can't load compiled
+# imports
+vet: binaries
 	@echo "+ $@"
 	@go vet ./...
 
@@ -70,5 +72,3 @@ binaries: ${PREFIX}/bin/registry ${PREFIX}/bin/registry-api-descriptor-template 
 clean:
 	@echo "+ $@"
 	@rm -rf "${PREFIX}/bin/registry" "${PREFIX}/bin/registry-api-descriptor-template"
-
-	

--- a/circle.yml
+++ b/circle.yml
@@ -75,6 +75,11 @@ test:
     - gvm use stable && go version
     # - gvm use bleed && go version
 
+  # First thing: build everything. This will catch compile errors, and it's
+  # also necessary for go vet to work properly (see #807).
+    - gvm use stable && godep go install ./...:
+        pwd: $BASE_STABLE
+
   # FMT
     - gvm use stable && test -z "$(gofmt -s -l . | grep -v Godeps/_workspace/src/ | tee /dev/stderr)":
         pwd: $BASE_STABLE

